File: examples/server/routes/languages/dart/yon.dart
import 'dart_language_service.dart';

class Handler {
  static final DartLanguageService _service = DartLanguageService();
  static final Map<String, Map<String, Object?>> _responses = {
    '429': {'code': '429', 'detail': 'too many requests'},
    '431': {'code': '431', 'detail': 'header too large'},
    '451': {'code': '451', 'detail': 'unavailable legal'},
    '500': {'code': '500', 'detail': 'internal error'},
    '501': {'code': '501', 'detail': 'not implemented'},
  };

  static Future<Map<String, Object?>> DELETE(Map<String, dynamic> request) async {
    final response = _responses[_statusCode(request)];
    if (response != null) {
      return response;
    }

    return await _service.delete(request);
  }

  static String _statusCode(Map<String, dynamic> request) {
    final query = request['query'];
    final raw = query is Map ? query['code'] : null;
    return raw is num ? raw.floor().toString() : (raw?.toString() ?? '');
  }
}

File: examples/server/routes/languages/cpp/yon.cpp
#include "CppLanguageService.cpp"

class Handler {
public:
    static YonJson GET(const YonJson& request) {
        const std::string code = statusCode(request);
        if (code == "502") return response("502", "bad gateway");
        if (code == "503") return response("503", "service unavailable");
        if (code == "504") return response("504", "gateway timeout");
        if (code == "505") return response("505", "http version not supported");
        if (code == "506") return response("506", "variant also negotiates");

        return service().describe(request);
    }

private:
    static CppLanguageService& service() {
        static CppLanguageService instance;
        return instance;
    }

    static std::string statusCode(const YonJson& request) {
        const YonJson* query = request.get("query");
        if (query == nullptr) return "";
        const YonJson* code = query->get("code");
        return code == nullptr ? "" : code->asString();
    }

    static YonJson response(const std::string& code, const std::string& detail) {
        return YonJson::object({
            {"code", code},
            {"detail", detail},
        });
    }
};
